### ISSUE TITLE
Handling of tool errors (e.g. Database)

### ISSUE THREAD

**Issue by mattlindsey** [patterns-ai-core/langchainrb#432]
Example below shows where a Tool raises an error, but the Assistant doesn't recognize it and happily gives wrong answer.   Maybe the Assistant can at least acknowledge that an error happened, and in the specific case below if the Assistant gets access to a 'schema' function/tool, it should work better.

![image](https://github.com/andreibondarev/langchainrb/assets/5638339/2d2ad856-431a-439e-bf9c-1280c4aa7ccc)

**Comment by mattlindsey** [comment]
Would making sure any error message is put on the Thread help?  I don't see it anywhere above except the log message.

**Comment by mattlindsey** [comment]
This looks better:
```
 #<Langchain::Message:0x0000000105a35ab0
  @content=
   "PG::UndefinedColumn: ERROR:  column \"name\" does not exist\nLINE 1: SELECT COUNT(*) FROM users WHERE LENGTH(name) > 5\n                                                ^\n",
  @role="tool",
  @tool_call_id="call_yKmQDXW3OpJM28m8JO7IowAy",
  @tool_calls=[]>,
 #<Langchain::Message:0x0000000105beed98
  @content=
   "I apologize for the confusion. It seems that the column \"name\" does not exist in the users table. Could you please provide the correct column name for the user's name?",
  @role="assistant",
  @tool_call_id=nil,
  @tool_calls=[]>]
```

Return error below in database.rb.  Maybe in other tools too?
```ruby
    #
    # Evaluates a sql expression
    #
    # @param input [String] sql expression
    # @return [Array] results
    #
    def execute(input:)
      Langchain.logger.info("Executing \"#{input}\"", for: self.class)

      db[input].to_a
    rescue Sequel::DatabaseError => e
      Langchain.logger.error(e.message, for: self.class)
      e.message     ###### ADDED THIS #######
    end
   ```

**Comment by mattlindsey** [comment]
Fixed by #743
